### PR TITLE
Set up farming opportunities balances and redirects

### PR DESCRIPTION
**Pull request by bootsboogie**
#11

### CODE REVIEW

File: src/hooks/useContractMethods.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useActiveProvider } from './useActiveProvider'
+import { useContract } from './useContract'
+import { FOX_ETH_FARMING_ADDRESS } from 'lib/constants'
+import farmAbi from 'abis/farmingAbi.json'
+
+export const useContractMethods = (contractAddress?: string) => {
```
Comment (unclerom) [inline src/hooks/useContractMethods.ts:6]: should you allow passing in the abi as well to make this a more generic hook for using contract methods?

Comment (bootsboogie) [reply src/hooks/useContractMethods.ts:6]: Do the farming contracts have different abi's? Should I update the stakingContracts constant to include an abi as well?

Comment (unclerom) [reply src/hooks/useContractMethods.ts:6]: currently no, the v2 will have the same abi. but when reading the name of the hook it seemed to be a generic hook for any contract methods, i guess thats what led to me making that comment

Comment (bootsboogie) [reply src/hooks/useContractMethods.ts:6]: Oh ok yea I dont know if we actually even need that anymore any way I might just get rid of it.

Comment (bootsboogie) [reply src/hooks/useContractMethods.ts]: Updated

**Review by unclerom — APPROVED** [review]

**Review by reallybeard — APPROVED** [review]

**Pull request merged**